pallets/schemas/src/lib.rs

```diff
@@ -402,19 +485,46 @@ pub mod pallet {
 			model_type: ModelType,
 			payload_location: PayloadLocation,
 			settings: BoundedVec<SchemaSetting, T::MaxSchemaSettingsPerSchema>,
+			schema_name_version: Option<(SchemaNamePayload, SchemaVersion)>,
 		) -> Result<SchemaId, DispatchError> {
 			let schema_id = Self::get_next_schema_id()?;
+			let (schema_name, version) = match schema_name_version {
+				Some((name_payload, schema_version)) => {
+					let parsed_name = SchemaName::try_parse::<T>(name_payload)?;
+					(parsed_name, schema_version)
+				},
+				None => {
```

Comment:
**Question:** maybe we should remove deprecated extrinsics instead of auto assign a name?

Reply:
I don't think we should auto-assign a name, but I do think we should auto-increment the version if a name is supplied.
Actually, I think `schema_version` should not be part of the payload, but be an auto-increment--however, this might involve an extra storage read to get the last version. I don't think that's too bad, though, since creating a schema will not be a heavily-used extrinsic.

Will we still support unnamed schemas? I don't necessarily see any reason not to...

Reply:
I think unnamed schemas are ok. Testing schemas etc... don't need an official name.

Either way I think we will need a way for governance to name (although not re-name) an existing schema.

Reply:
Auto incrementing versions would be good and I'm ok with schema creation being a fairly expensive operation

Reply:
Ok then, I'll take a look into auto-incrementing the version approach. 
If we consider having a name as **optional** for schemas then I think for existing well known schemas we can apply names during migration but to allow assigning names for future unassigned schemas we will need a new extrinsic (only accessible via governance)